### PR TITLE
[2.11.x] DDF-3426 Close client sockets in HttpSolrClientFactory

### PR DESCRIPTION
**Pull request by blen-desta**
Backport of https://github.com/codice/ddf/pull/2602
#### What does this PR do?
Closes client sockets in HttpSolrClientFactory

#### Who is reviewing it? 
@brjeter @Lambeaux 

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@lessarderic
@pklinef 

#### How should this be tested? (List steps with links to updated documentation)
Full build.
Install, ingest, queries, and save/read a workspace.

#### What are the relevant tickets?
[DDF-3426](https://codice.atlassian.net/browse/DDF-3426)

### CODE REVIEW

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java [hunk 1]
```diff
@@ -281,7 +284,9 @@ private static SSLContext getSslContext() {
         | KeyStoreException
         | KeyManagementException e) {
       throw new IllegalArgumentException(
-          "Unable to use javax.net.ssl.keyStorePassword to load key material to create SSL context for Solr client.");
+          "Unable to use "
+              + KEY_STORE_PASS
```
Comment (coyotesqrl) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java]: ✏️ This particular change seems a little silly to me. I don't know how I feel about breaking out an exception message to insert a const in the middle of it.

**Comment by coyotesqrl** [comment]
✏️ I would suggest that heroing this also involve an install, ingest of data, queries, and save/read of a workspace. That should exercise Solr sufficiently to ensure that the cores have started successfully.

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
PR build failed because it was unable to load `org.sonar.scanner.scan.ProjectLock` and `org.sonar.scanner.scan.DefaultInputModuleHierarchy`. Rebuild ...

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
Owasp failure:
`One or more dependencies were identified with vulnerabilities that have a CVSS score greater then '7.0': solr-core-6.6.1.jar: CVE-2017-12629`

**Comment by blen-desta** [comment]
build now

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java [hunk 2]
```diff
@@ -249,7 +254,7 @@ private static CloseableHttpClient getSecureHttpClient(boolean retryRequestsOnEr
   private static SSLContext getSslContext() {
     if (System.getProperty("javax.net.ssl.keyStore") == null
```
Comment (lessarderic) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java]: ❓ Should we create constants for these hard-coded strings as well?

**Comment by blen-desta** [comment]
Pushed changes. Building again.

**Comment by blen-desta** [comment]
build now

**Comment by AzGoalie** [comment]
Herod on windows 👍

**Comment by clockard** [comment]
Manually verified test steps. 👍

**Pull request merged**